setup.py

```diff
@@ -69,6 +70,35 @@
     shutil.rmtree(stale_egg_info)
 
 
+# IMPORTANT:
+# 1. all version requirements should be in this table
+# 2. once modified, run: python setup.py --name to generate src/transformers/dependency_versions_table.py
+_deps = [
+    "flax==0.2.2",
+    "jax>=0.2.0",
+    "jaxlib==0.1.55",
+    "numpy",
+    "python>=3.6.0",
+    "regex!=2019.12.17",
+    "tokenizers==0.9.4",
+    "tqdm>=4.27",
+]
+
+# tokenizers: "tokenizers==0.9.4" lookup table
+# support non-versions file too so that they can be checked at run time
+deps = { b:a for a,b in (re.findall(r"^(([^!=<>]+)(?:[!=<>].*)?$)", x)[0] for x in _deps) }
+
+def deps_list(*pkgs):
+    return [deps[pkg] for pkg in pkgs]
+
+with open("src/transformers/dependency_versions_table.py", "w") as f:
+    entries = "\n".join([f'    "{k}": "{v}",' for k,v in deps.items()])
+    f.write("""# THIS FILE HAS BEEN AUTOGENERATED. To update:
```

Comment:
Could this lead to problems in environments where there is no write permission?

Reply:
Please examplify? setup.py is almost always writing something to the filesystem.

Reply:
True -> this should be fine

Reply:
I was more just thinking out loud :-) I think you're 100% right. This should not pose a problem!